modules/custom/social_scroll/src/Form/SocialScrollSettingsForm.php

```diff
@@ -0,0 +1,160 @@
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The SocialScrollManager manager.
+   *
+   * @var \Drupal\social_scroll\SocialScrollManager
+   */
+  protected $socialScrollManager;
+
+  /**
+   * SocialScrollSettingsForm constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory service.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $moduleHandler
+   *   The module handler service.
+   * @param \Drupal\social_scroll\SocialScrollManager $social_scroll_manager
+   *   The SocialScrollManager manager.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ModuleHandlerInterface $moduleHandler, SocialScrollManager $social_scroll_manager) {
+    parent::__construct($config_factory);
+    $this->moduleHandler = $moduleHandler;
+    $this->socialScrollManager = $social_scroll_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    /* @noinspection  PhpParamsInspection */
+    return new static(
+      $container->get('config.factory'),
+      $container->get('module_handler'),
+      $container->get('social_scroll.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'social_scroll_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [self::CONFIG_NAME];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $all_views = array_map(function ($view_id) {
+      return str_replace('views.view.', '', $view_id);
+    }, $this->configFactory->listAll('views'));
+
+    $blocked_views = $this->socialScrollManager->getBlockedViewIds();
+    $views = array_diff($all_views, $blocked_views);
+    $config = $this->config(self::CONFIG_NAME);
+
+    $form['page_display'] = [
+      '#type' => 'item',
+      '#title' => $this->t('Here\'s a list of views that have a "page" display.'),
+    ];
+
+    $form['settings']['button_text'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button Text'),
+      '#default_value' => $config->get('button_text'),
+      '#maxlength' => '255',
+    ];
+
+    $form['settings']['automatically_load_content'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Automatically Load Content'),
+      '#description' => $this->t('Automatically load subsequent pages as the user scrolls.'),
+      '#default_value' => $config->get('automatically_load_content'),
+    ];
+
+    $form['settings']['views'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Views'),
+    ];
+
+    $options = [];
+    foreach ($views as $view) {
+      $current_view = $this->configFactory->getEditable($this->socialScrollManager->getConfigName($view));
+      $label = $current_view->getOriginal('label');
+
+      if ($label) {
```

Comment:
This check isn't needed because the label is a mandatory field

Reply:
In some cases, this label can be NULL.
I added this conditions because faced with an error.